### ISSUE TITLE
Unit tests failing for python3.10

### ISSUE THREAD

**Issue by gabrielcocenza** [juju/charm-helpers#712]
Unit tests are failing for the python3.10 as it is possible to see [here](https://pipelines.actions.githubusercontent.com/serviceHosts/10f50de8-d7e6-4d79-899e-ce657c5c81cc/_apis/pipelines/1/runs/235/signedlogcontent/5?urlExpires=2022-05-27T18%3A34%3A57.5315033Z&urlSigningMethod=HMACV1&urlSignature=LN5nn%2FGv%2F0sEb3c2VunpT0vUmLDMb1EUky4LPtX8wsA%3D) and in this [pastebin](https://pastebin.canonical.com/p/wFKkpZ8DVm/).

Apparently, the sequence of the tests that run has changed from one python version to another and this might break some tests.

**Comment by ajkavanagh** [comment]
> Apparently, the sequence of the tests that run has changed from one python version to another and this might break some tests.

So, this probably means that the tests are not properly isolated.  All three errors look like either a mock failure of some kind.

**Comment by gabrielcocenza** [comment]
> So, this probably means that the tests are not properly isolated.  All three errors look like either a mock failure of some kind.

That is right. I just looked at the `test_ceph_context_ec_pool_no_rbd_pool` and the problem there is that is missing the env var for `JUJU_UNIT_NAME` that was happening on test `test_save_scriptrc`. Now that the sequence of tests changed, setting this env var on the test that is failing is necessary.

**Comment by ajkavanagh** [comment]
> That is right. I just looked at the test_ceph_context_ec_pool_no_rbd_pool and the problem there is that is missing the env var for JUJU_UNIT_NAME that was happening on test test_save_scriptrc. Now that the sequence of tests changed, setting this env var on the test that is failing is necessary.

Indeed, the correct thing to do is to ensure that all of the tests are isolated so that run order does not affect the tests.  Some tests may be leaking which means that mocking is not done in those (currently) not failing tests.
